dali/pipeline/operators/geometric/flip_test.cc

```diff
@@ -0,0 +1,133 @@
+
+struct TestTlData {
+  std::vector<float> _data;
+
+  explicit TestTlData(void *data_ptr) noexcept
+  : _data(2 * data_width * data_height * data_channels) {
+    auto data_size = data_width * data_height * data_channels * sizeof(float);
+    std::memcpy(_data.data(), data_ptr, data_size);
+    std::memcpy(_data.data() + data_size / sizeof(float), data_ptr, data_size);
+  }
+
+  float *data() {
+    return _data.data();
+  }
+};
+
+TestTlData nhwc_tensor_list_data(&data_nhwc[0][0]);
+
+float data_nchw[2][2][3][2][3] = {{{{{1.1, 2.1, 3.1}, {4.1, 5.1, 6.1}},
+                                    {{1.2, 2.2, 3.2}, {4.2, 5.2, 6.2}},
+                                    {{1.3, 2.3, 3.3}, {4.3, 5.3, 6.3}}},
+                                   {{{4.1, 5.1, 6.1}, {1.1, 2.1, 3.1}},
+                                    {{4.2, 5.2, 6.2}, {1.2, 2.2, 3.2}},
+                                    {{4.3, 5.3, 6.3}, {1.3, 2.3, 3.3}}}},
+                                  {{{{3.1, 2.1, 1.1}, {6.1, 5.1, 4.1}},
+                                    {{3.2, 2.2, 1.2}, {6.2, 5.2, 4.2}},
+                                    {{3.3, 2.3, 1.3}, {6.3, 5.3, 4.3}}},
+                                   {{{6.1, 5.1, 4.1}, {3.1, 2.1, 1.1}},
+                                    {{6.2, 5.2, 4.2}, {3.2, 2.2, 1.2}},
+                                    {{6.3, 5.3, 4.3}, {3.3, 2.3, 1.3}}}}};
+
+TestTlData nchw_tensor_list_data(&data_nchw[0][0]);
+
+class FlipTest : public testing::DaliOperatorTest {
+  GraphDescr GenerateOperatorGraph() const override {
+    GraphDescr graph("Flip");
+    return graph;
+  }
+};
+
+std::vector<Arguments> arguments = {
+    {{"horizontal", 0}, {"vertical", 0}},
+    {{"horizontal", 1}, {"vertical", 0}},
+    {{"horizontal", 0}, {"vertical", 1}},
+    {{"horizontal", 1}, {"vertical", 1}}
+};
+
+std::vector<Arguments> devices = {
+    {{"device", std::string{"cpu"}}},
+    {{"device", std::string{"gpu"}}},
+};
+
+std::vector<Arguments> layout = {
+    {{"nhwc", true}},
+    {{"nhwc", false}}
+};
+
+void FlipVerify(TensorListWrapper input, TensorListWrapper output, Arguments args) {
+  int _horizontal = args["horizontal"].GetValue<int>();
+  int _vertical = args["vertical"].GetValue<int>();
+  std::string dev = args["device"].GetValue<std::string>();
+  auto output_d = output.CopyTo<CPUBackend>();
+  auto item_size = output_d->type().size();
+  for (size_t i = 0; i < output_d->ntensor(); ++i) {
+    auto size =
+        output_d->tensor_shape(i)[0] * output_d->tensor_shape(i)[1] * output_d->tensor_shape(i)[2];
+    auto out_tensor = output_d->raw_tensor(i);
+    if (output_d->GetLayout() == DALI_NHWC)
+      ASSERT_EQ(std::memcmp(out_tensor, &data_nhwc[_horizontal][_vertical], size * item_size), 0);
+    else if (output_d->GetLayout() == DALI_NCHW)
+      ASSERT_EQ(std::memcmp(out_tensor, &data_nchw[_horizontal][_vertical], size * item_size), 0);
+  }
+}
+
+TEST_P(FlipTest, NhwcTest) {
```

Comment:
`NhwcTest` doesn't seem to be a good name for this since it is instantiated for both NHWC and NHWC